tools/shared.py

```diff
@@ -1168,13 +1168,100 @@ def __setattr__(self, attr, value):
 class Settings(object):
   __metaclass__ = Settings2
 
+# llvm-ar appears to just use basenames inside archives. as a result, files with the same basename
+# will trample each other when we extract them. to help warn of such situations, we warn if there
+# are duplicate entries in the archive
+def warn_if_duplicate_entries(archive_contents, archive_filename_hint=''):
```

Comment:
is this and the next method new code? (new changes to behavior?) or is this just moving code around for multiprocessing purposes? (if so where is the removal?)